content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,399 @@
+
+Features include:
+* Validate source definitions
+* Authenticate to sources (SSH, user/password, API token)
+* Validate source authenticity (PGP)
+* Detect source changes based on update policies (semver)
+* Fetch resources both on-demand (via webhooks) and on-a-schedule (at a configured polling interval)
+* Package the fetched resources into a well-known format (tar.gz, yaml)
+* Make the artifacts addressable by their source identifier (sha, version, ts)
+* Make the artifacts available in-cluster to interested 3rd parties (such as the Kustomize Controller and Helm Controller)
+* Notify interested 3rd parties of source changes and availability (status conditions, events, hooks)
+
+### Secret Decryption via SOPS
+
+The [Kustomize Controller will decrypt Secret values encrypted using Mozilla's SOPS CLI][Mozilla SOPS Guide]
+with OpenPGP, AWS KMS, GCP KMS or Azure Key Vault and stored in the source Git repository as Kubernetes Secret manifests.
+The encrypted secret can be safely stored in a public or private Git repository.
+
+The Kustomize Controller will pull the Kubernetes Secret manifest with encrypted values from the Source, (the metadata
+is stored in plain-text), then Kustomize Controller decrypts its encrypted values using the supplied key.
+
+Note, it's a good idea to also back up your secret values in a password manager or other form of secure external storage.
+
+The decrypted manifests are kept in memory and passed on to the next stage.
+
+### Kustomize Controller builds and validates resources
+
+The Kustomize Controller runs the go library equivalent of a `kustomize build` against the `Kustomization.spec.path` to recursively generate and render (or inflate) any Kustomize overlays. (All manifests are passed through Kustomize, even those that don't include a `kustomization.yaml`.)
+
+Before it applies YAML or JSON resurce declarations to the Kubernetes API for its cluster, the Kustomize Controller
+reads the artifact files from its source path and builds them using the Kustomize Go library's `build` call. This call
+returns any custom resource definitions (CRDs), namespaces, or other cluster-wide resources it renders before
+subordinate custom resources or namespace-scoped resources so that they will be available in the API for the resources
+that refer to or use them.
+
+* FAQ: [How does Flux run `kustomize build` internally?][Kustomize Build Flags] (Replicate Kustomize behavior locally with the Kustomize CLI)
+
+### Kustomize Controller applies changes with Server-Side Apply
+
+**Needs a link to FluxCD documentation for Server-Side Apply process!** _The only current on-point reference is the [Server-side reconciliation is coming](https://fluxcd.io/blog/2021/09/server-side-reconciliation-is-coming/) blog post.  This would likely be considered a gap in FluxCD project docs._
```

Comment:
```suggestion
**Needs a link to FluxCD documentation for Server-Side Apply process!** _The only current on-point reference is the [Server-side reconciliation is coming](/blog/2021/09/server-side-reconciliation-is-coming/) blog post.  This would likely be considered a gap in FluxCD project docs._
```